### ISSUE TITLE
Unify pretty print parse errors

### ISSUE THREAD

**Issue by GleasonK** [openxla/stablehlo#63]
### Request description

Per feedback on #38.

_"As I'm looking through StablehloOps.cpp, I see that we have a bunch of different ways to express expected vs actual mismatches: "expected, saw", "expected, got", "expected, found", etc. We'll need to sort this out at some later point."_

This issue will be used to track the unification of parse error messages.

**Comment by burmako** [comment]
@zhouxin913 You might be interested in this ticket as well, given our discussion about verification and shape inference error messages from earlier today.

**Comment by zhouxin913** [comment]
Yes, aligning errors message will not only improve user experience, but also make code clean and easy to maintain. Some quick ideas in verifiers/shape functions plan like providing macros and/or utils functions to wrap some errors for example: *Expect dim i of tensor A to be same as dim j of tensor B, but got 1!=2*. It's quite similar to this issue for print parse. Will share a quick design and looking forward to work on this together.

**Comment by zhouxin913** [comment]
Update: this request is related to https://github.com/openxla/stablehlo/issues/154, which is work in progress addressing similar problem for type inference. Consider start this work once that one is addressed for code sharing.
